### PR TITLE
[Feature] Support DDL [Drop external catalog catalog_name;] for external catalog

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

related with https://github.com/StarRocks/starrocks/issues/5824

### CODE REVIEW

**Comment by adzfolc** [comment]
The `DropCatalogStmt.toSql()` func would temporary have `EXTERNAL` field, and it will be changed to `INTERNAL or EXTERNAL` when internal catalog is supported.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropCatalogStmt.java|14|18|77.78%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return catalogName;
+    }
+
+    public void analyze() throws SemanticException {
+        // TODO check permission
+        if (Strings.isNullOrEmpty(catalogName)) {
+            throw new SemanticException("'catalog name' can not be null or empty");
+        }
+        try {
+            FeNameFormat.checkCatalogName(catalogName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:30]: There is no need to check format again when dropping catalog

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
+        return visitor.visitDropCatalogStatement(this, context);
+    }
+
+    @Override
+    public String toSql() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("DROP EXTERNAL CATALOG ");
+        sb.append(catalogName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:49]: catalogName -> 'catalogName'

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+package com.starrocks.analysis;
+
+import com.google.common.base.Strings;
+import com.starrocks.catalog.InternalCatalog;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.FeNameFormat;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.AstVisitor;
+
+public class DropCatalogStmt extends DdlStmt {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:12]: You could try to implement processing dropCatalogStmt in `DdlExecutor.java`

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java (hunk 2 above)
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:49]: how about name?

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java (hunk 1 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java]: Format check has been removed.

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java (hunk 2 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java]: > how about name?

Do you mean rename `String catalogName;` to `String name;` ?

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java]: Yes. It is confusing in this context.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropCatalogStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.analysis;
+
+import com.google.common.base.Strings;
+import com.starrocks.catalog.InternalCatalog;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.AstVisitor;
+
+public class DropCatalogStmt extends DdlStmt {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:10]: Add some annotations for future support of internal catalog?

**Review by caneGuy — DISMISSED** [review]
other LGTM

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropCatalogStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.analysis;
+
+import com.google.common.base.Strings;
+import com.starrocks.catalog.InternalCatalog;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.AstVisitor;
+
+// ToDo: to support internal catalog in the future
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/DropCatalogStmt.java:10]: should use this format
// TODO(zhuodong):

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropCatalogStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|2|2|100.00%|

**Review by caneGuy — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/analysis/CatalogStmtTest.java [hunk 6]
```diff
@@ -53,6 +53,18 @@ public void testSelectNonReservedCol() {
...
+    @Test
+    public void testDropCatalogParserAndAnalyzer() {
+        String sql_1 = "DROP EXTERNAL CATALOG catalog_1";
+        StatementBase stmt = AnalyzeTestUtil.analyzeSuccess(sql_1);
+        Assert.assertTrue(stmt instanceof DropCatalogStmt);
+        String sql_2 = "DROP EXTERNAL CATALOG";
+        AnalyzeTestUtil.analyzeFail(sql_2);
+        String sql_3 = "DROP EXTERNAL CATALOG default";
+        AnalyzeTestUtil.analyzeFail(sql_3);
+        Assert.assertEquals("DROP EXTERNAL CATALOG 'catalog_1'", stmt.toSql());
```
[the remark this answers is not in the document]
Comment (adzfolc) [reply fe/fe-core/src/test/java/com/starrocks/analysis/CatalogStmtTest.java:66]: > DROP EXTERNAL CATALOG 'catalog_name'

added

**Review by stephen-shelby — APPROVED** [review]

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropCatalogStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|2|2|100.00%|

**Review by caneGuy — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**